api/controllers/zetter.ts

```diff
@@ -1,296 +1,306 @@
-import tweets from '../data/tweets';
-import users from '../data/users';
-import followers from '../data/followers';
-import favorities from '../data/favorities';
 import { sign } from 'jsonwebtoken';
 import { createHash } from 'crypto';
 import { Request, Response } from 'express';
-import { FavoriteType, FollowerType, ProfileType, TweetType } from '../utils/types';
+import { prisma } from '../utils/prisma';
+import { Tweet, User, Favorite } from '@prisma/client';
 
-export const getTweets = (req: Request, res: Response) => {
+export const getTweets = async (req: Request, res: Response) => {
     const limit: number = 10;
-    const followingUsers: Array<FollowerType> = followers.filter((follower) => follower.from === req.user.id);
-    const followingUsersAndMyIds: Array<number> = followingUsers.map((obj) => obj.to);
-    let followingUsersTweetsList: Array<TweetType> = [];
-
-    followingUsersAndMyIds.push(req.user.id);
-    tweets.forEach((tweet) => {
-        if (followingUsersAndMyIds.includes(tweet.createdBy)) {
-            followingUsersTweetsList.push(tweet);
-        }
+    const followings = await prisma.user.findMany({
+        where: {
+            followedBy: {
+                some: { id: req.user.id },
+            },
+        },
     });
-
-    followingUsersTweetsList.sort(function (a: TweetType, b: TweetType) {
-        return Number(b.createdAt) - Number(a.createdAt);
+    const timelineTweetsList: Array<Tweet> = await prisma.tweet.findMany({
+        where: {
+            createdBy: {
+                in: [req.user.id, ...followings.map((following) => following.id)],
+            },
+        },
```

Comment:
ごめんよく考えたら、そもそもfollowingsをいったんとってくる必要もなくて、これで一発でとれそう
```suggestion
        where: {
            user: {
                OR: [
                    { id: req.user.id },
                    { followedBy: { some: { id: req.user.id } } }
                ]
            }
        }
```